bins/src/bin/datadog-static-analyzer-server.rs

```diff
@@ -58,13 +58,25 @@ async fn serve_static(
     server_configuration: &State<ServerConfiguration>,
     name: &str,
 ) -> Option<NamedFile> {
-    let name_optional = server_configuration.files_to_serve.get(name);
+    server_configuration.static_directory.as_ref()?;
 
-    if let Some(name) = name_optional {
-        NamedFile::open(name).await.ok()
-    } else {
-        None
+    if name.contains("..") {
+        return None;
     }
+
+    if name.starts_with('.') {
+        return None;
+    }
+
+    let full_path = Path::new(
+        server_configuration
+            .static_directory
+            .clone()
+            .unwrap()
+            .as_str(),
+    )
+    .join(Path::new(name));
```

Comment:
No need to use `Path::new` here. The `AsRef` trait will make it work for a `&str`.